docs/basic-features/environment-variables.md

```diff
@@ -124,3 +124,7 @@ This one is useful when running tests with tools like `jest` or `cypress` where
 There is a small difference between `test` environment, and both `development` and `production` that you need to bear in mind: `.env.local` won't be loaded, as you expect tests to produce the same results for everyone. This way every test execution will use same env defaults across different executions by ignoring your `.env.local` (which is intended to override the default set).
 
 > **Note**: similar to Default Environment Variables, `.env.test` file should be included in your repository, but `.env.test.local` shouldn't, as `.env*.local` are intended to be ignored through `.gitignore`.
+
+## NODE_ENV
+
+The value of `NODE_ENV` is used to determined which `.env.*` file to load when running the application. However, this value is not passed from the environment to your application code when using the `next` CLI. Instead, when running `next dev` then `process.env.NODE_ENV` is set to `development`, and when running `next build` it is set to `production`.
```

Comment:
I find this confusing: 

> The value of `NODE_ENV` is used to determined which `.env.*` file to load when running the application. However, this value is not passed from the environment to your application code when using the `next` CLI

We don't need to mention any of that, instead let's add a note saying that Next.js will take care of setting `NODE_ENV` for you.

Start by updating the CLI docs to add a note under each script (`next dev` - `next build` - `next start`) saying this:

> `NODE_ENV` will be set to `development` by Next.js when you run this command

(The above would be an example for the `next dev` docs)

Once that's done, you can replace this title with a note like this one:

> **Note:** Next.js will take care of setting `NODE_ENV` for you based on the environment, check out the [CLI docs](/docs/api-reference/cli) for more details.

That sound way better for me, users shouldn't be worrying about `NODE_ENV` details as it never has to be touched

Reply:
Yeah that makes more sense :+1 this PR was definitely a bit of a straw man given I've very rarely written tech docs for a wide range of users, so thanks for the input!

What you've said makes sense apart from "based on the environment" in your last example, which I think would read better if it was change to "based on the `next` command you're using" or something similar.

The reason being is that I think "the environment" has a much wider connotation and could literally be interpreted as "based on some opaque environment settings", when actually it's much simpler than that.

Reply:
Thinking about it, the above doesn't quite address the issue that I had where running `next start` from inside Jest loads the `.env.test` environment, put sets `NODE_ENV` to `production`.

If you state

> `NODE_ENV` will be set to`production` by Next.js when you run this command

for prod, it might make it all the more confusing that it will load the `.env.*` file based on the `NODE_ENV` from the outer environment.

Perhaps you don't feel that's worth addressing but I know I'm not the only one to have experienced that problem and if it is worth addressing then I'm not quite sure how you get around it _without_ mentioning `NODE_ENV`.

Reply:
@reicheltp To make this simpler, let's start with the CLI docs and then update the environment variables docs accordingly. Include what you think should be there and I'll make more suggestions 👍